File: scratch/analyze_db.mjs
import pkg from 'pg';
const { Client } = pkg;
import fs from 'fs';
import dotenv from 'dotenv';

dotenv.config({ path: '.env.local' });

async function analyze() {
    const client = new Client({
        connectionString: process.env.DATABASE_URL,
        ssl: { rejectUnauthorized: false }
    });

    try {
        await client.connect();
        console.log("Connected successfully!");

        let report = "# Análisis de Tablas de Supabase\n\n";

        // Obtener tablas publicas
        const resTables = await client.query(`
            SELECT table_name 
            FROM information_schema.tables 
            WHERE table_schema = 'public' 
            AND table_type = 'BASE TABLE'
            ORDER BY table_name;
        `);

        for (const row of resTables.rows) {
            const t = row.table_name;
            report += `## Tabla: ${t}\n`;

            // Contar filas
            const resCount = await client.query(`SELECT COUNT(*) FROM "${t}"`);
            const count = resCount.rows[0].count;
            report += `*Total de filas: ${count}*\n\n`;

            // Obtener columnas
            const resCols = await client.query(`
                SELECT column_name, data_type 
                FROM information_schema.columns 
                WHERE table_name = $1 
                AND table_schema = 'public'
                ORDER BY ordinal_position;
            `, [t]);

            report += `| Columna | Tipo |\n`;
            report += `|---|---|\n`;
            for (const col of resCols.rows) {
                report += `| ${col.column_name} | \`${col.data_type}\` |\n`;
            }
            report += "\n";

            // Obtener 2 resultados de muestra
            if (parseInt(count) > 0) {
                try {
                    const resSample = await client.query(`SELECT * FROM "${t}" LIMIT 2`);
                    report += `**Muestra de datos:**\n\`\`\`json\n${JSON.stringify(resSample.rows, null, 2)}\n\`\`\`\n\n`;
                } catch (e) {
                    report += `*Error obteniendo muestra: ${e.message}*\n\n`;
                }
            }
        }

        fs.writeFileSync('scratch/db_report.md', report, 'utf-8');
        console.log("Report generated at scratch/db_report.md");
    } catch (e) {
        console.error("Connection error:", e.message);
    } finally {
        await client.end();
    }
}

analyze();

File: scratch/analyze_db.py
import psycopg2
from psycopg2 import sql

import psycopg2
from psycopg2 import sql

# Parametros extraidos de la cadena: postgresql://postgres.cdinoxmyukpasihkkkmd:Jo.9839514500@aws-0-us-west-2.pooler.supabase.com:6543/postgres
DB_USER = "postgres.cdinoxmyukpasihkkkmd"
DB_PASS = "Jo.9839514500"
DB_HOST = "aws-0-us-west-2.pooler.supabase.com"
DB_PORT = "6543"
DB_NAME = "postgres"

def analyze():
    try:
        conn = psycopg2.connect(
            user=DB_USER,
            password=DB_PASS,
            host=DB_HOST,
            port=DB_PORT,
            database=DB_NAME,
            sslmode="require"
        )
        cur = conn.cursor()
        
        # Get all tables in public schema
        cur.execute("""
            SELECT table_name 
            FROM information_schema.tables 
            WHERE table_schema = 'public' 
            AND table_type = 'BASE TABLE'
            ORDER BY table_name;
        """)
        tables = [row[0] for row in cur.fetchall()]
        
        print(f"Total tables found: {len(tables)}\n")
        
        results = []
        for table in tables:
            # Get row count
            cur.execute(sql.SQL("SELECT COUNT(*) FROM {}").format(sql.Identifier(table)))
            count = cur.fetchone()[0]
            
            # Get columns
            cur.execute("""
                SELECT column_name, data_type 
                FROM information_schema.columns 
                WHERE table_name = %s 
                AND table_schema = 'public'
                ORDER BY ordinal_position;
            """, (table,))
            columns = cur.fetchall()
            
            results.append({
                "name": table,
                "count": count,
                "columns": columns
            })
            
        # Get views
        cur.execute("""
            SELECT table_name 
            FROM information_schema.views 
            WHERE table_schema = 'public'
            ORDER BY table_name;
        """)
        views = [row[0] for row in cur.fetchall()]
        
        print("TABLES ANALYSIS:")
        for res in results:
            print(f"- {res['name']} ({res['count']} rows)")
            # col_str = ", ".join([f"{c[0]} ({c[1]})" for c in res['columns']])
            # print(f"  Columns: {col_str}")
            
        print("\nVIEWS FOUND:")
        for view in views:
            print(f"- {view}")
            
        cur.close()
        conn.close()
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    analyze()
